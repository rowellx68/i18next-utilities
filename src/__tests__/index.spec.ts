import { describe, it, expect } from 'vitest';
import { build } from 'vite';
import plugin from '../index';

describe('vite-plugin-typed-i18next-loader', () => {
  it('should generate type definitions for default locale', async () => {
    const result = await build({
      build: {
        lib: {
          entry: './src/__tests__/__fixtures__/entrypoint.ts',
          formats: ['es']
        },
        outDir: './src/__tests__/__fixtures__/dist'
      },
      plugins: [plugin({
        namespaceResolution: 'basename',
        defaultLocale: 'en-GB',
        output: './src/__tests__/__fixtures__/types/i18next.d.ts',
        paths: ['./src/__tests__/__fixtures__/locales/'],
      })],
    }) as any[];

    const output = result[0].output[0]

    expect(output.code).toMatchSnapshot();
  });
});
