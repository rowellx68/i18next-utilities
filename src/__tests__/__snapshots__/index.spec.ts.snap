// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite-plugin-typed-i18next-loader > should generate type definitions for default locale 1`] = `
"const e = { en: { namespace: { message: "Hello, World!" }, translation: { "app.title": "Hello World" } }, "en-GB": { namespace: { message: "Hello, World!", other: "This is a test" }, translation: { "app.title": "Hello World" } } };
console.log(e);
"
`;
